syntax = "proto3";

package lyra;

option go_package = "internal/pb";

service WhisperService {
    rpc Transcribe(TranscribeRequest) returns (TranscribeResponse);
    rpc HealthCheck(HealthCheckRequest) returns (HealthCheckResponse);
    rpc CreateTranscriptionTask(CreateTranscriptionTaskRequest) returns (CreateTaskResponse);
    rpc GetTaskStatus(GetTaskStatusRequest) returns (GetTaskStatusResponse);
}

message TranscribeRequest {
    string task_id = 1;
    string file_key = 2;
}

message CreateTranscriptionTaskRequest {
    string task_id = 1;
    string file_key = 2;
}

message TranscribeResponse {
    string text = 1;
    string error = 2;
}

message HealthCheckRequest {}
message HealthCheckResponse {
    string status = 1;
}

message CreateTaskResponse {
    string task_id = 1;
    string error = 2;
}

message GetTaskStatusRequest {
    string task_id = 1;
}

message GetTaskStatusResponse {
    string status = 1;
    string result = 2;
    string error = 3;
    int64 created_at = 4;
    int64 updated_at = 5;
}
